lesson16/main.js

```diff
@@ -0,0 +1,119 @@
+"use strict";
+
+const jsonUrl = "https://jsondata.okiba.me/v1/json/yFXrE210903000829";
+const newsWrapper = document.getElementById("js-news");
+
+const createElementWithClassName = (element, name) => {
+  const variableName = document.createElement(element);
+  variableName.classList.add(name);
+  return variableName;
+};
+
+//comment-icon
+const commentIcon = createElementWithClassName("span", "comment-icon");
+
+//new-label
+const newLabel = createElementWithClassName("span", "new-label");
+newLabel.textContent = "NEW!";
+
+//news-img
+const imgWrapper = createElementWithClassName("div", "news__img");
+
+
+//tab
+const createNewTab = data => {
+  const tabUl = document.querySelector(".news__tabs");
+  const tabFragment = document.createDocumentFragment();
+
+  data.forEach((key) => {
+    const tabLi = createElementWithClassName("li", "tab");
+    tabLi.textContent = key.category;
+    tabLi.dataset.tab = key.id;
+    if (key.selected === true) {
+      tabLi.classList.add("is-active");
+    }
+
+    tabLi.addEventListener("click", tabSwitch);
+    tabFragment.appendChild(tabLi);
+  });
+
+  tabUl.appendChild(tabFragment);
+};
+
+//news
+const createNewContent = data => {
+  const contentFragment = document.createDocumentFragment();
+  
+  data.forEach( key => {
+    const newsContent = createElementWithClassName("div", "news__content");
+    const contentUl = createElementWithClassName("ul", "news__lists");
+    const titleFragment = document.createDocumentFragment();
+    newsContent.dataset.content = key.id;
+
+    key.article.forEach( array => {
+      const contentLi = createElementWithClassName("li", "news__item");
+      const a = document.createElement("a");
+      a.textContent = array.title;
+      a.href = "#";
+      titleFragment.appendChild(contentLi).appendChild(a);
+    })
+    
+    if (key.selected === true) {
+      newsContent.classList.add("is-show");
+    }
+    contentFragment.appendChild(newsContent).appendChild(contentUl).appendChild(titleFragment);
+  });
+
+  newsWrapper.appendChild(contentFragment);
+};
+
+//tabSwitch
+const tabSwitch = () => {
+  const tabs = document.querySelectorAll("[data-tab]");
+  const contents = document.querySelectorAll("[data-content]");
+
+  const handleClick = e => {
+    e.preventDefault();
+    const targetTab = e.target;
+    const targetTabVal = targetTab.dataset.tab;
+    
+    let i = 0;
+    while (i < tabs.length) {
+      contents[i].classList.remove("is-show");
+      tabs[i].classList.remove("is-active");
+      i++;
+    };
+    
+    newsWrapper.querySelectorAll('[data-content="' + targetTabVal + '"]')[0].classList.add("is-show");
```

Comment:
```suggestion
   const target = `[data-content="${targetTabVal}"]`
    newsWrapper.querySelectorAll(target)[0].classList.add("is-show");
```

Reply:
It's spread syntax! I haven't used it so much so far, it's helpful! Thank you.